src/pamiq_core/data/impls/random_replacement_buffer.py

```diff
@@ -21,21 +22,41 @@ def __init__(
         self,
         collecting_data_names: Iterable[str],
         max_size: int,
-        replace_probability: float = 1.0,
+        replace_probability: float | None = None,
+        expected_survival_length: int | None = None,
     ) -> None:
         """Initialize a RandomReplacementBuffer.
 
         Args:
             collecting_data_names: Names of data fields to collect.
             max_size: Maximum number of data points to store.
             replace_probability: Probability of replacing an existing element when buffer is full.
-                Must be between 0.0 and 1.0 inclusive. Default is 1.0 (always replace).
+                Must be between 0.0 and 1.0 inclusive. If None and expected_survival_length is provided,
+                this will be computed automatically. Default is 1.0 if both are None.
+            expected_survival_length: Expected number of steps that data should survive in the buffer.
+                Used to automatically compute replace_probability if replace_probability is None.
+                Cannot be specified together with replace_probability.
 
         Raises:
-            ValueError: If replace_probability is not between 0.0 and 1.0 inclusive.
+            ValueError: If replace_probability is not between 0.0 and 1.0 inclusive, or if both
+                replace_probability and expected_survival_length are specified.
         """
         super().__init__(collecting_data_names, max_size)
 
+        if replace_probability is None:
+            if expected_survival_length is None:
+                replace_probability = 1.0
+            else:
+                replace_probability = (
+                    self.compute_replace_probability_from_expected_survival_length(
+                        max_size, expected_survival_length
+                    )
+                )
+        elif expected_survival_length is not None:
+            raise ValueError(
+                "Cannot specify both replace_probability and expected_survival_length. "
+                "Please specify only one of them."
+            )
         if not (1.0 >= replace_probability >= 0.0):
```

Comment:
inclusiveなので
```suggestion
        if not (1.0 > replace_probability > 0.0):
```
になりませんか

Reply:
すみません間違えました